syntax = "proto3";
import "google/protobuf/empty.proto";

package open_greenery.rpc.relay;

service Relay {
  rpc SetConfig(Config) returns (google.protobuf.Empty) {}
  rpc ManualControl(ManualControlRequest) returns (google.protobuf.Empty) {}
  rpc SetMode(ModeSetting) returns (google.protobuf.Empty) {}
  rpc GetRelayStatus(google.protobuf.Empty) returns (RelayStatus) {}
  rpc GetServiceStatus(google.protobuf.Empty) returns (ServiceStatus) {}
}

message Config {
  // msecs since start of day
  int32 day_start = 1;
  int32 day_end = 2;
}

message ManualControlRequest {
  enum Control {
    CONTROL_ENABLE = 0;
    CONTROL_DISABLE = 1;
    CONTROL_TOGGLE = 2;
  }
  Control control = 1;
}

message ModeSetting {
  enum Mode {
    MODE_MANUAL = 0;
    MODE_AUTO = 1;
  }
  Mode mode = 1;
}

message RelayStatus {
  bool is_enabled = 1;
}

message ServiceStatus {
  ModeSetting mode_settings = 1;
  RelayStatus relay_status = 2;
  Config config = 3;
}
